File: system/files/FDFile.java
package common.system.files;

import common.CommonStatic;
import common.pack.Context.ErrType;
import common.system.fake.FakeImage;

import java.io.BufferedInputStream;
import java.io.File;
import java.io.FileInputStream;
import java.io.InputStream;
import java.nio.file.Files;

public class FDFile implements FileData {

	private final File file;

	public FDFile(File f) {
		file = f;
	}

	@Override
	public byte[] getBytes() {
		byte[] bs = new byte[(int) file.length()];

        try (BufferedInputStream buf = new BufferedInputStream(Files.newInputStream(file.toPath()))) {
			buf.read(bs, 0, bs.length);
			return bs;
        } catch (Exception e) {
            e.printStackTrace();
			return null;
        }
	}

	@Override
	public FakeImage getImg() {
		return FakeImage.read(file);
	}

	@Override
	public InputStream getStream() {
		return CommonStatic.ctx.noticeErr(() -> new FileInputStream(file), ErrType.ERROR,
				"failed to read bcuzip at " + file);
	}

	@Override
	public int size() {
		return (int) file.length();
	}

	@Override
	public String toString() {
		return file.getName();
	}

}

File: system/files/FDFile.kt
package common.system.files

import common.CommonStatic
import common.pack.Context.ErrType
import common.pack.Context.SupExc
import common.system.fake.FakeImage
import common.util.Data
import java.io.*

class FDFile(private val file: File) : FileData {
    override fun getBytes(): ByteArray? {
        return try {
            val bs = ByteArray(file.length().toInt())
            val buf = BufferedInputStream(FileInputStream(file))
            buf.read(bs, 0, bs.size)
            buf.close()
            bs
        } catch (e: IOException) {
            e.printStackTrace()
            null
        }
    }

    override fun getImg(): FakeImage {
        return Data.Companion.err<FakeImage>(SupExc<FakeImage> { FakeImage.Companion.read(file) })
    }

    override fun getStream(): InputStream {
        return CommonStatic.ctx.noticeErr<FileInputStream>(SupExc<FileInputStream> { FileInputStream(file) }, ErrType.ERROR,
                "failed to read bcuzip at $file")
    }

    override fun size(): Int {
        return file.length().toInt()
    }

    override fun toString(): String {
        return file.name
    }
}
